### PR TITLE
docs: Remove references to deprecated types

### PR DESCRIPTION
**Pull request by christoshrousis**
As per the [1.6 release](https://deno.com/blog/fresh-1.6#simplified-typings), and commentary by @csvn [here](https://github.com/denoland/fresh/issues/2108#issuecomment-1836625789), this PR aims to replace the references to deprecated typing in the documentation.

Please see inline commentary for parts I'm unclear on.

### CODE REVIEW

File: docs/latest/concepts/app-wrapper.md [hunk 1]
```diff
@@ -37,12 +37,12 @@ export default function App({ Component, state }: AppProps) {
 
 Similar to routes and layouts, the app wrapper can be made asynchronous. This
 changes the function signature so that the first argument is the `Request`
-instance and the second one is the `AppContext`.
+instance and the second one is the `FreshContext`.
 
 ```tsx routes/_app.tsx
-import { AppContext } from "$fresh/server.ts";
```
Comment (christoshrousis) [inline docs/latest/concepts/app-wrapper.md:43]: In the generated code for a new 1.6 fresh project, we get

`import { type AppContext } from "$fresh/server.ts";`

It is unclear whether the Deno/fresh team prefer `import { type` over `import type` syntax, and whether the docs should reflect this.

Comment (marvinhagemeister) [reply docs/latest/concepts/app-wrapper.md:43]: I'm neutral on that as both styles are merely a hint to the runtime that the import can be stripped.

**Review by marvinhagemeister — APPROVED** [review]
Nice, thank you for making this PR! The changes look good to me 👍

File: docs/latest/concepts/middleware.md [hunk 2]
```diff
@@ -18,15 +18,15 @@ properties, e.g. `ctx.state.loggedIn = true`, but you can also replace the
...
 interface State {
   data: string;
 }
 
 export async function handler(
   req: Request,
-  ctx: MiddlewareHandlerContext<State>,
+  ctx: FreshContext<State>,
 ) {
   ctx.state.data = "myData";
```
Comment (christoshrousis) [inline docs/latest/concepts/middleware.md:31]: The release documentation mentions:

`RouteContext (for async routes/layouts/app wrapper)` 

When I tried to confirm this, and use that typing in this code example, I got a typing error:

`Property 'next' does not exist on type 'RouteContext<State>'.deno-ts(2339)`

**Pull request merged**